.github/workflows/pull-request.yml

```diff
@@ -63,3 +70,31 @@ jobs:
             ${{ runner.os }}-yarn-
       - name: Lint project
         uses: vtex/action-lint@master
+
+  tsDeclarations:
+    name: TypeScript Declarations
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-node@master
+        with:
+          node-version: 12.x
+        env:
+          RUNNER_TEMP: /tmp
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: 'echo "::set-output name=dir::$(yarn cache dir)"'
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: '${{ steps.yarn-cache-dir-path.outputs.dir }}'
+          key: '${{ runner.os }}-yarn-${{ hashFiles(''**/yarn.lock'') }}'
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install dependencies
+        run: |
+          yarn install --frozen-lockfile
+          yarn --cwd react install --frozen-lockfile
+      - name: Declaration emit
+        run:
+          yarn tsc -p react/tsconfig.json --emitDeclarationOnly -d --declarationDir types
```

Comment:
Why do we need this? So other apps can import their typings?

Reply:
yes, builder-hub wasn't generating the typings for this app due to an error in one file (but it didn't fail the build) so I though it could be a good idea to make sure this won't happen again in the future

Reply:
wouldn't be better to rebuild it after fixing the error? I mean, if this isn't very specific to this repo, we'll need to replace this CI everywhere and ignore builder-hub, correct?

Reply:
> wouldn't be better to rebuild it after fixing the error?

I don't think I understood the question 🤔

I fixed the error and added this CI so it doesn't happen again, and so you are asking that we should include this in builder-hub itself?

Reply:
> I fixed the error and added this CI so it doesn't happen again

It's specific to this repository only?

Reply:
If yes, lgtm, but otherwise, I'm worried that we'll need to replicate this to other repositories, and in this scenario, I'd rather have this bug fixed at the more appropriate place (builder hub perhaps)

Reply:
> It's specific to this repository only?

I believe it could happen with other repos as well, although I've only seen it on this one

> If yes, lgtm, but otherwise, I'm worried that we'll need to replicate this to other repositories, and in this scenario, I'd rather have this bug fixed at the more appropriate place (builder hub perhaps)

oh, understood! yeah, I think this should live in builder-hub as well (at least as a warning)

Reply:
so you think I should remove this?

Reply:
Damn, I was pretty sure I had answered this :((

But yes, I think it would be better to remove if the problem is already solved for now (from what I understood, you have added this so it won't repeat in the future, r8?). We should also communicate the responsables of builder-hub about this problem. Looks fair to you?

By the way, what is the problem exactly?

Reply:
your suggestion looks good to me! I'll remove it then

I didn't understand much of the problem, but what I figured out is that we use the `Context` type for `OrderItemsContext`, but it isn't exported, so typescript can't generate the declarations because we were using a "private" type. Adding the export to this `Context` type fixed it :-)

Reply:
I see, gj!